### ISSUE TITLE
.Net 8 not installable on Ubuntu 24.04

### ISSUE THREAD

**Issue by PavelCibulka** [dotnet/sdk#39907]
Hello, it is not possible to install .Net 8 on Ubuntu 24.04 due to Ubuntu having libicu74. (.Net 8 can use libicu up to version 0.72 - libicu72). Please add libicu74 as dependency alternative for libicu.

**installing dotnet:**
sudo apt install dotnet-sdk-8.0
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 dotnet-runtime-deps-8.0 : Depends: libicu but it is not installable or
                                    libicu72 but it is not installable or
                                    libicu71 but it is not installable or
                                    libicu70 but it is not installable or
                                    libicu69 but it is not installable or
                                    libicu68 but it is not installable or
                                    libicu67 but it is not installable or
                                    libicu66 but it is not installable or
                                    libicu65 but it is not installable or
                                    libicu63 but it is not installable or
                                    libicu60 but it is not installable or
                                    libicu57 but it is not installable or
                                    libicu55 but it is not installable or
                                    libicu52 but it is not installable
E: Unable to correct problems, you have held broken packages.

**Comment by baronfel** [comment]
@leecow heads-up, we may have a packaging dependencies issue

**Comment by richlander** [comment]
Are you using our `jammy` feed on Noble?

**Comment by PavelCibulka** [comment]
It was from 23.10. When I try 24.04 it says that package is not available:
sudo nano /etc/apt/sources.list.d/microsoft-prod.list
deb [arch=amd64,arm64,armhf signed-by=/usr/share/keyrings/microsoft-prod.gpg] https://packages.microsoft.com/ubuntu/24.04/prod noble main

pavel@pavel-yoga:~$ sudo apt install dotnet-sdk-8.0
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package dotnet-sdk-8.0 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'dotnet-sdk-8.0' has no installation candidate

**Comment by chriscarpenter12** [comment]
I'm seeing similar failures from Apt trying to fetch the sdk in our Packer builds as commented [here](https://github.com/dotnet/core/issues/9224#issuecomment-2025975208)

Seems it's only `dotnet-sdk-8.0` that's having the issue for Ubuntu 22.04 for me

**Comment by MichaelSimons** [comment]
cc @NikolaMilosavljevic

**Comment by NikolaMilosavljevic** [comment]
@PavelCibulka I was not able to repro this issue. I used a fresh Docker container with `ubuntu:noble` image.

The following steps were successful:
```
apt update
apt install dotnet-sdk-8.0
dotnet --info
```

Do you still get the same error? Can you share specifics about the machine environment where this was attempted, and full set of repro steps?

**Comment by NikolaMilosavljevic** [comment]
> I'm seeing similar failures from Apt trying to fetch the sdk in our Packer builds as commented [here](https://github.com/dotnet/core/issues/9224#issuecomment-2025975208)
> 
> Seems it's only `dotnet-sdk-8.0` that's having the issue for Ubuntu 22.04 for me

@chriscarpenter12 was this an intermittent issue in your machine environment? I am not able to reproduce it. If you are still experiencing this issue, can you share more specifics and steps to reproduce?

**Comment by chriscarpenter12** [comment]
@NikolaMilosavljevic It was intermittent but over a few days and through the weekend. Seems to be fine since late Monday. I haven’t had a build fail since. Thanks for following up

**Comment by NikolaMilosavljevic** [comment]
Closing as no-repro.

**Comment by cspwizard** [comment]
> @PavelCibulka I was not able to repro this issue. I used a fresh Docker container with `ubuntu:noble` image.
> 
> The following steps were successful:
> 
> ```
> apt update
> apt install dotnet-sdk-8.0
> dotnet --info
> ```
> 
> Do you still get the same error? Can you share specifics about the machine environment where this was attempted, and full set of repro steps?

I've the same issue. In you example the dotnet is installed from archive.ubuntu.com/... repo, in the original example by @PavelCibulka  packages.microsoft.com/ubuntu/24.04/prod reoo is used. And it still doesn't have any dotnet packages published. 
Are there any reference date when the packages will be available?

**Comment by richlander** [comment]
Please read the plan, one comment above. Microsoft won't be publishing 24.04 packages.

**Comment by mikevankuik** [comment]
I had the same issue.
The solution for me was to comment out the line in the microsoft-prod.list file. (I can possibly remove it, haven´t tried it yet I like to continue developing.)
![image](https://github.com/dotnet/sdk/assets/29262650/354c9b1a-92e5-45a5-a884-f719b7a40440)

**Comment by forReason** [comment]
Can confirm that removing it resolved the issue for me
```
sudo rm /etc/apt/sources.list.d/microsoft-prod.list
```

**Comment by sergiomartinezsbf** [comment]
works for me, thanks @forReason

**Comment by lngr** [comment]
In case someone stumbles above the same issue, here's a further cause:  I had upgraded from 23.10 to 24.04 today and could not install `dotnet-sdk-8.0` even though I removed `microsoft-prod.list` from the sources list as suggested by @forReason 

However, after one hour of scratching my head I noticed that I also had added an entry to `/etc/apt/preferences` to deselect any dotnet packages from the official Ubuntu sources in the past.  After removing that entry, I was successfully able to install `dotnet-sdk-8.0`

**Comment by artgmrs** [comment]
works for me too, thanks @forReason

**Comment by dotarvid** [comment]
The solutions provided above do not work for me. I am using Ubuntu 24.04.

I solved it by removing `microsoft-ubuntu-focal-prod.list` and microsoft-ubuntu-jammy-prod.list`.

```bash
sudo rm -rf /etc/apt/sources.list.d/microsoft-ubuntu-jammy-prod.list
sudo rm -rf /etc/apt/sources.list.d/microsoft-ubuntu-focal-prod.list
```

I hope this helps.

**Comment by N-Endy** [comment]
Thanks @forReason 
Your solution works perfectly

**Comment by 10950761** [comment]
The solution worked perfectly for me

**Comment by vurso** [comment]
This is a problem on the latest version of Ubuntu and if you upgrade to `Ubuntu 24.04 LTS x86_64` with an existing dotnet 8 SDK install that will also break.

I have tried everything stated above doesn't work and also trying the backports to install dotnet 7 SDK doesn't actually install the SDK only the runtimes.

Its a shame I have to downgrade my Ubuntu installation not fun means starting over with a fresh install I wish I had known about this issue beforehand.

**Comment by jkoritzinsky** [comment]
We've heard that the upgrade scenario is painful. We're taking a look at ways to provide a better transition experience here (tracked by https://github.com/dotnet/runtime/issues/105046).

**Comment by vurso** [comment]
Interestingly this worked for me:  https://www.tecmint.com/upgrade-ubuntu-22-04-to-24-04/  - I didn't need to do the final step `sudo do-release-upgrade -d` the upgrade through this method worked just fine.

I then installed dotnet 8 SDK as follows:

Remove dotnet packages

`sudo apt remove dotnet* aspnetcore* netstandard*`

`sudo apt update && sudo apt install -y dotnet-sdk-8.0`

After a successful install I reloaded the bash session with `source ~/.bashrc` and then typed `dotnet -h` which showed the SDK commands and I was then able to successfully create a quick test console app, build and run.

Hope this helps someone else.

In terms of upgrading I went from 22.04 LTS to 24.04 LTS and the upgrade process doesn't take long (i5 laptop with 32GB ram and SSD drive).
